s3transfer/tasks.py

```diff
@@ -245,6 +157,97 @@ def _get_all_main_kwargs(self):
             kwargs[key] = result
         return kwargs
 
+    def _submit_task(self, executor, task):
+        future = executor.submit(task)
+        # Add this created future to the list of associated future just
+        # in case it is needed during cleanups.
+        self._transfer_coordinator.add_associated_future(future)
+        return future
+
+
+class SubmissionTask(Task):
+    """A base class for any submission task
+
+    Submission tasks are the top-level task used to submit a series of tasks
+    to execute a particular transfer.
+    """
+    def _main(self, transfer_future, **kwargs):
+        """
+        :type transfer_future: s3transfer.futures.TransferFuture
+        :param transfer_future: The transfer future associated with the
+            transfer request that tasks are being submitted for
+
+        :param kwargs: Any additional kwargs that you may want to pass
+            to the _submit() method
+        """
+        try:
+            # Before submitting any tasks, run all of the on_queued callbacks
+            on_queued_callbacks = get_callbacks(transfer_future, 'queued')
+            for on_queued_callback in on_queued_callbacks:
+                on_queued_callback()
+
+            # Once callbacks have been ran set the status to running.
+            self._transfer_coordinator.set_status_to_running()
```

Comment:
Is this right?  The transfer coordinator's status of running is for the actual transfer task right (i.e the GetObject() from S3).  Shouldn't the actual transfer task be the one that flips the status to running?